src/client.ts

```diff
@@ -4236,7 +4237,10 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * @returns Promise which resolves: an empty object
      * @returns Rejects: with an error response.
      */
-    public setAccountData(eventType: EventType | string, content: IContent): Promise<{}> {
+    public setAccountData<K extends keyof AccountDataEvents>(
```

Comment:
there is also `setRoomAccountData` -- shouldn't we change that too?

Reply:
That'd for a future PR to keep the diffs smaller